pkg/cmd/server/kubernetes/master/controller.go

```diff
@@ -145,3 +149,40 @@ func (c *SchedulerControllerConfig) RunController(ctx kubecontroller.ControllerC
 
 	return true, nil
 }
+
+// NB: this is funky -- it's actually a Kubernetes controller, but we run it as an OpenShift controller in order
+// to get a handle on OpenShift clients, so that our delegating scales getter can work.
+
+type HorizontalPodAutoscalerControllerConfig struct {
+	HeapsterNamespace string
+	HeapsterScheme    string
+	HeapsterService   string
+	HeapsterPort      string
+}
+
+func (c *HorizontalPodAutoscalerControllerConfig) RunController(ctx oscontroller.ControllerContext) (bool, error) {
+	hpaClient := ctx.ClientBuilder.ClientOrDie("horizontal-pod-autoscaler")
+	hpaOriginClient := ctx.ClientBuilder.DeprecatedOpenshiftClientOrDie("horizontal-pod-autoscaler")
```

Comment:
open an issue to switch to the generated client.  I want to delete the old client in 3.7